plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1332,6 +1333,36 @@ public OptionalLong delete(ConnectorSession session, JdbcTableHandle handle)
         }
     }
 
+    @Override
+    public OptionalLong update(ConnectorSession session, JdbcTableHandle handle, Map<String, Domain> assignments)
+    {
+        checkArgument(handle.isNamedRelation(), "Unable to update from synthetic table: %s", handle);
+        checkArgument(handle.getLimit().isEmpty(), "Unable to update when limit is set: %s", handle);
+        checkArgument(handle.getSortOrder().isEmpty(), "Unable to update when sort order is set: %s", handle);
+        List<JdbcColumnHandle> columns = getColumns(session, new JdbcTableHandle(((JdbcNamedRelationHandle) handle.getRelationHandle()).getSchemaTableName(), ((JdbcNamedRelationHandle) handle.getRelationHandle()).getRemoteTableName(), Optional.empty()));
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            verify(connection.getAutoCommit());
+            PreparedQuery preparedQuery = queryBuilder.prepareUpdateQuery(
+                    this,
+                    session,
+                    connection,
+                    handle.getRequiredNamedRelation(),
+                    handle.getConstraint(),
+                    getAdditionalPredicate(handle.getConstraintExpressions(), Optional.empty()),
+                    assignments,
+                    columns);
+            try (PreparedStatement preparedStatement = queryBuilder.prepareStatement(this, session, connection, preparedQuery, Optional.empty())) {
+                int affectedRowsCount = preparedStatement.executeUpdate();
+                // In getPreparedStatement we set autocommit to false so here we need an explicit commit
+                connection.commit();
```

Comment:
hmmm, I see a `verify(connection.getAutoCommit())` above which suggests it's `true` when we reach here.